src/prefect/_internal/concurrency/services.py

```diff
@@ -113,18 +115,28 @@ async def _run(self):
 
     async def _main_loop(self):
         while True:
-            item: T = await self._queue.get()
+            thing = await self._queue.get()
 
-            if item is None:
+            if thing is None:
                 break
 
+            item, context = thing
+
+            tokens: Dict[ContextVar, Token] = {}
+            for key, value in context.items():
+                token = key.set(value)
+                tokens[key] = token
```

Comment:
I was imagining this as a separate utility — okay with just using it here though

Reply:
:+1: I can make it a context manager.